In [ ]:
import pandas as pd

dir_in_train = [f"C:/Users/hoang/Documents/Dataset_KLTN/ciciot2023_extracted/merge-processed/Incremental_1.3/session{i}_train.parquet" for i in range(0, 3)]
dir_in_test = [f"C:/Users/hoang/Documents/Dataset_KLTN/ciciot2023_extracted/merge-processed/Incremental_1.3/session{i}_test.parquet" for i in range(0, 3)]

all_rows = []

all_labels = set()

# Quét tất cả file train + test để lấy tất cả nhãn
for train_path in dir_in_train:
    test_path = train_path.replace("_train.parquet", "_test.parquet")

    df_train = pd.read_parquet(train_path)
    df_test = pd.read_parquet(test_path)

    all_labels.update(df_train["Label"].unique())
    all_labels.update(df_test["Label"].unique())

all_labels = sorted(all_labels)  # sắp xếp alphabet cho dễ nhìn

rows = []

df1 = pd.read_parquet(dir_in_train[2])
keys= df1['Label'].value_counts()
labels =list(keys.keys())

print(labels)
keys = {0: 'Benign', 1:'DDoS', 2: 'DoS', 3: 'Reconnaisance', 4: 'Vulnerability_Scan'}


for train_path in dir_in_train:
    name = train_path.split("/")[-1].replace("_train.parquet", "")
    test_path = train_path.replace("_train.parquet", "_test.parquet")

    df_train = pd.read_parquet(train_path)
    df_test = pd.read_parquet(test_path)

    train_counts = df_train["Label"].value_counts()
    test_counts = df_test["Label"].value_counts()

    row = {"File": name}

    # Nhóm Train
    for lbl in all_labels:
        row[f"Train: {keys[lbl]}"] = train_counts.get(lbl, 0)

    # Nhóm Test
    for lbl in all_labels:        
        row[f"Test: {keys[lbl]}"] = test_counts.get(lbl, 0)

    rows.append(row)

df = pd.DataFrame(rows)
print(df)
print(df.to_markdown(index=False))


[2, 1, 3, 0, 4]
       File  Train: Benign  Train: DDoS  Train: DoS  Train: Reconnaisance  \
0  session0         340007       339707      340286                     0   
1  session1         113411       113300      113344                 42444   
2  session2          84867        85020       85150                 85013   

   Train: Vulnerability_Scan  Test: Benign  Test: DDoS  Test: DoS  \
0                          0         59993       60293      59714   
1                          0         19922       20033      19989   
2                      42450         15133       14980      14850   

   Test: Reconnaisance  Test: Vulnerability_Scan  
0                    0                         0  
1                 7556                         0  
2                14987                      7550  
| File     |   Train: Benign |   Train: DDoS |   Train: DoS |   Train: Reconnaisance |   Train: Vulnerability_Scan |   Test: Benign |   Test: DDoS |   Test: DoS |   Test: Reconnaisance |   Test:

: 

| File     |   Train: Benign |   Train: DDoS |   Train: DoS |   Train: Reconnaisance |   Train: Vulnerability_Scan |   Test: Benign |   Test: DDoS |   Test: DoS |   Test: Reconnaisance |   Test: Vulnerability_Scan |
|:---------|----------------:|--------------:|-------------:|-----------------------:|----------------------------:|---------------:|-------------:|------------:|----------------------:|---------------------------:|
| session0 |          340007 |        339707 |       340286 |                      0 |                           0 |          59993 |        60293 |       59714 |                     0 |                          0 |
| session1 |          113411 |        113300 |       113344 |                  42444 |                           0 |          19922 |        20033 |       19989 |                  7556 |                          0 |
| session2 |           84867 |         85020 |        85150 |                  85013 |                       42450 |          15133 |        14980 |       14850 |                 14987 |                       7550 |